src/facadeUI/LogInManager.java

```diff
@@ -28,45 +48,63 @@ public void login(String id) {
 		// if isLoggedOn = true -> error
 		if (isLoggedOn) {
 			// error handle
+			UIUpdater.error("You are already logged on. Logout and try again.");
 		} else {
 			// verify with DB and get UID
-			// uid = DBUID
-			// on error = display error to UI
-			uid = 0;
-			isLoggedOn = true;
+			int userID;
+			try {
+				userID = cModel.getIdFromEmail(id);
+				uid = userID;
+				isLoggedOn = true;
+				// get the matches
+				List<Customer> matched = cModel.getMatchedCustomers(uid);
+				ArrayList<String> matches = new ArrayList<String>();
+				for (Customer match : matched) {
+					String s = "Name: " + match.getName() + "\n" + "uid: " + match.getCustomerId();
+					matches.add(s);
+				}
+				UIUpdater.login(id, matches);
+			} catch (SQLException e) {
+				UIUpdater.error("Error logging in: "+ e.getMessage());
+			}
 		}
 	}
 
 	// SignUp Method
 	public void signUp(String email, String name, int pid) {
-		// TODO
 		if (isLoggedOn) {
-			// Error: please logout before you sign up
+			UIUpdater.error("Please logout before signing up!");
+		} else {
+			// SignUp Process
+			try {
+				cModel.createCustomer(email, name, pID);
+				// Auto SignOn
```

Comment:
Consider calling MatchModel.createMatches here to generate matches between the new user with existing users